syntax = "proto3";

import "pizzeria/ingredient.proto";

package feedme;

// Define the RecipeActions enum
enum RecipeActions {
    UNKNOWN = 0; // Default value
    CHOP = 1; // Chop action
    MIX = 2; // Mix action
    BAKE = 3; // Bake action
    BOIL = 4; // Boil action
    FRY = 5; // Fry action
    // Add more actions as needed
}

message Instruction
{
    string name = 1; // Name of the instruction
    repeated Ingredient ingredients = 2; // Ingredients involved in the instruction
    RecipeActions action = 3; // Action to be performed
    int64 duration = 4; // Duration of the instruction in seconds
}

message Recipe
{
    string name = 1; // Name of the recipe
    repeated Ingredient ingredients = 2; // List of ingredients
    int32 shelf_life = 3; // Shelf life in days
    int32 servings = 4; // Number of servings
    float base_cost = 5; // Base cost of the recipe
    float margin = 6; // Margin percentage, corrected spelling
    int64 preparation_time = 7; // Preparation time in seconds
    int64 cook_time = 8; // Cooking time in seconds
}
